include/kernels/gauge_random.cuh

```diff
@@ -0,0 +1,113 @@
+#include <quda_matrix.h>
+#include <gauge_field_order.h>
+#include <index_helper.cuh>
+#include <random_helper.h>
+#include <kernel.h>
+
+namespace quda {
+
+  template <typename Float_, int nColor_, QudaReconstructType recon_, bool group_>
+  struct GaugeGaussArg : kernel_param<> {
+    using Float = Float_;
+    using real = typename mapper<Float>::type;
+    static constexpr int nColor = nColor_;
+    static constexpr QudaReconstructType recon = recon_;
+    static constexpr bool group = group_;
+
+    using Gauge = typename gauge_mapper<Float, recon>::type;
+
+    int E[4]; // extended grid dimensions
+    int X[4]; // true grid dimensions
+    int border[4];
+    Gauge U;
+    RNGState *rng;
+    real sigma; // where U = exp(sigma * H)
+
+    GaugeGaussArg(const GaugeField &U, RNGState *rng, double sigma) :
+      kernel_param(dim3(U.LocalVolumeCB(), 2, 1)),
+      U(U),
+      rng(rng),
+      sigma(sigma)
+    {
+      int R = 0;
+      for (int dir = 0; dir < 4; ++dir) {
+        border[dir] = U.R()[dir];
+        E[dir] = U.X()[dir];
+        X[dir] = U.X()[dir] - border[dir] * 2;
+        R += border[dir];
```

Comment:
Arguably not the scope of this PR, but I think this is sufficiently simple --- we can get rid of `R`, it doesn't seem to be used anywhere. I checked `lib/gauge_random.cu` and confirmed this _wasn't_ a transcription issue.

Reply:
91eb8903e14293ce355417d050afd9a566542e1e